capstone-backend/src/main/java/com/google/sps/servlets/BookServlet.java

```diff
@@ -47,91 +32,25 @@ public BookServlet() throws IOException {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<Book> result = new ArrayList<>();
-    if (request.getParameter("id") != null) {
-      DocumentReference docRef = books.document(request.getParameter("id"));
-      ApiFuture<DocumentSnapshot> future = docRef.get();
-      DocumentSnapshot document = null;
-      Book book = null;
-      try {
-        document = future.get();
-        if (document.exists()) {
-          book = document.toObject(Book.class);
-        } else {
-          System.err.println("Error: no such document!");
-        }
-      } catch (Exception e) {
-        System.err.println("Error: " + e);
-      }
-      result.add(book);
+    List<Book> retrievedBooks = Utility.get(books, request, response, new GenericClass(Book.class));
```

Comment:
Nice - the methods on this servlet look really clean and concise with the introduction of the `Utility` methods!

Reply:
Thanks!